### PR TITLE
Refactor suggestion-internal-modules

### PR DESCRIPTION
**Pull request by c-bata**
**What this PR does / why we need it**:

* Avoid shadows of built-in methods (type, min, max, list).
* Fix comparison operator.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
None

**Special notes for your reviewer**:

None.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/suggestion/v1alpha3/internal/search_space.py [hunk 1]
```diff
@@ -50,12 +50,12 @@ def convertParameter(p):
 
 
 class HyperParameter(object):
-    def __init__(self, name, type, min, max, list, step):
+    def __init__(self, name, type_, min_, max_, list_, step):
```
Comment (c-bata) [inline pkg/suggestion/v1alpha3/internal/search_space.py]: In Python, These are common practice to avoid shadows of built-in methods and variables.

Comment (gaocegege) [reply pkg/suggestion/v1alpha3/internal/search_space.py]: LGTM

**Comment by c-bata** [comment]
Thank you for your swift review :smile:

OK. I'll submit some refactoring pull requests before working on #1100.

**Comment by johnugeorge** [comment]
CI seems to be slow now

**Pull request merged**